edx_analytics_transformers/transformers/caliper/event_transformers/video_events.py

```diff
@@ -0,0 +1,288 @@
+"""
+Transformers for video interaction events.
+
+NOTE: Currently Open edX only emits legacy events for video interaction i.e.
+- load_video
+- play_video
+- stop_video
+- complete_video (proposed)
+- pause_video
+- seek_video
+
+These event names are to be changed following the new event name format.
+Currently, mobile apps emits these events using the new names but will be
+added in edx-platform too. Therefore, we are adding support for legacy event names
+as well as new names.
+
+These events:
+- edx.video.loaded
+- edx.video.played
+- edx.video.stopped
+- edx.video.paused
+- edx.video.position.changed
+- edx.video.completed (proposed)
+"""
+from datetime import timedelta
+
+from isodate import duration_isoformat
+
+from edx_analytics_transformers.transformers.caliper.transformer import CaliperTransformer
+from edx_analytics_transformers.transformers.caliper.registry import CaliperTransformersRegistry
+
+
+EVENTS_ACTION_MAP = {
+    'load_video': 'Retrieved',
+    'play_video': 'Started',
+    'stop_video': 'Ended',
+    'complete_video': 'Ended',
+    'pause_video': 'Paused',
+    'seek_video': 'JumpedTo',
+}
+
+
+def make_video_block_id(video_id, course_id, video_block_name='video+block', block_version='block-v1'):
+    """
+    Return formatted video block id for provided video and course.
+
+    Arguments:
+        video_id        (str) : id for the video object
+        course_id       (str) : course key string
+        video_block_name(str) : video block prefix to generate video id
+        block_version   (str) : xBlock version
+
+    Returns:
+        str
+    """
+    return '{block_version}:{course_id}+type@{video_block_name}@{video_id}'.format(
+        block_version=block_version,
+        course_id=course_id,
+        video_block_name=video_block_name,
+        video_id=video_id
+    )
+
+
+def convert_seconds_to_iso(seconds):
+    """
+    Convert seconds from integer to ISO format.
+
+    Arguments:
+        seconds (int): number of seconds
+
+    Return:
+        str
+    """
+    return duration_isoformat(timedelta(
+        seconds=seconds
+    ))
+
+
+class BaseVideoTransformer(CaliperTransformer):
+    """
+    Base transformer for video interaction events.
+    """
+    type = 'MediaEvent'
+
+    def get_action(self, current_event, _):
+        """
+        Return action for the caliper event.
+
+        Arguments:
+            current_event (dict):   untransformed event
+            _             (dict):   transformed event
+
+        Returns:
+            str
+        """
+        return EVENTS_ACTION_MAP[current_event['name']]
+
+    def get_object(self, current_event, caliper_event):
+        """
+        Return object for the caliper event.
+
+        Arguments:
+            current_event (dict):   untransformed event
+            caliper_event (dict):   transformed event
+
+        Returns:
+            str
+        """
+        self.json_load_event()
```

Comment:
In case of browser generated events, `data` field in the event has JSON stringified event. Here we are converting that Stringified JSON into python dict object.